### DISCUSSION TITLE
New parameters to pass row and col to calculateFormula

### DISCUSSION THREAD

**Discussion by emadebManyDesigns** [handsontable/hyperformula#1320] [Feature requests]
I have a Custom formula that generate the result using the row where it is called.

This is done using the value of state.formulaAddress.row.
```
export class BidEvoFunctionsPlugin extends FunctionPlugin {
  fieldVal(ast, state) {
    return this.runFunction(
      ast.args,
      state,
      this.metadata('VAL'),
      () => {
         const row = state && state.formulaAddress ? state.formulaAddress.row : 0;
         console.log(row);
         ...
      }
    );
  }
}
```

It works very well when the function is called by Handsontable.

I have also a field outside handsontable, where the user can write a new formula that will be inserted on handsontable.
To generate the preview of insertedFormula I call the calculateFormula function:
```
const formulaString = "=VAL()";
const res = hfInstance.calculateFormula(formulaString, sheetId);
```

The problem is that calculateFormula let only to specify the sheetId, but not the row and the col where the formula will be inserted, so my preview fails because it is always executed in the first row.

To fix that problem, I need the new parameters `row` and `col` in the calculateFormula function:
HyperFormula.ts
```
  public calculateFormula(formulaString: string, sheetId: number, coll: number, row: number): CellValue | CellValue[][] {
    ...
    const {ast, address, dependencies} = this.extractTemporaryFormula(formulaString, sheetId, col, row)
    ...
```
```
  private extractTemporaryFormula(formulaString: string, sheetId: number = 1, col: number = 0, row: number = 0): { ast?: Ast, address: SimpleCellAddress, dependencies: RelativeDependency[] } {
    ...
    const address = {sheet: sheetId, col: col, row: row}
    ...
```

It is possible to add that fix to the official HyperFormula release?

**Comment by sequba** [comment]
Hi @emadebManyDesigns, thanks for reaching out to us. It is a great idea for a feature. Currently, our core team prioritizes bug and interoperability fixes before any new features, so it is not on our immediate roadmap.

But, if you want to see this feature available in HyperFormula sooner, we are open to [contributions from the community](https://hyperformula.handsontable.com/guide/contributing.html), and we will be delighted to collaborate on it.

**Comment by emadebManyDesigns** [comment]
Hi.
I have created a pull request to modify the calculateFormula function.
https://github.com/handsontable/hyperformula/pull/1720
